geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java

```diff
@@ -498,6 +498,11 @@ public StoppableCountDownLatch getInitializationLatchAfterGetInitialImage() {
    */
   private final Lock clientMetaDataLock = new ReentrantLock();
 
+  /**
+   * Lock for updating the cache service profile for the region.
+   */
+  private final Lock cacheServiceProfileUpdateLock = new ReentrantLock();
```

Comment:
Can we move this Lock and the public acquire/release methods into LuceneServiceImpl.java since that's the only class that uses the Lock? In general, we should prefer not exposing any synchronization mutexes outside the code that directly uses it.

Reply:
@kirklund CacheProfileService was added a new feature and Lucene was the first to use it. Later on, as we create more features over geode, other modules may need to use this lockl. Also, multiple cacheProfiles (a mix of lucene and others) may get accessed at the same time hence this was put here. I will discuss this with Barry (he designed the cache profile) and get back.

Reply:
@kirklund  If fact as we add more cache profiles and multiple features use them, Local Region class will have to be refactored to use this lock more to get synchronized.

Reply:
Locks are now used within localRegion